Brady pythonify

brajac709 opened the pull request:
Hey I got kind of bored and started playing around with Rank Panda again.  One thing I found was that I was able to get it to work with python 2.7.  So I updated the README to reflect that.  I think the other changes I've made were fairly minor (but those were done a while ago...).  But I remember finding the python 2.6 download page was kind of difficult, plus its old and people should be using 2.7.

brajac709 committed:
Dialogs

List comprehensions and useful dialogs for unlocking.

brajac709 committed:
better python code and some other fixes

For ease of use, when adding waypoints, if a time is left blank it is
set to zero.

brajac709 committed:
Removed SplineGenerator Class

also started trying to make things iterative. still working on it....

brajac709 committed:
updated README

Adds instructions for use with Python 2.7

astory commented on RankPanda/Commands.py:
```diff
@@ -12,56 +12,64 @@ def __init__(self, length, beginLocation):
     # Be sure to overwrite this in each Command!
     def CalcLocation(self, count, beginLocation):
+        """ Given the beginning location, calculates the location that the
+         rank should be in after count counts.  Uses information such as
```
Nit: please line wrap to match the same column the initial """ starts on, and end with """ on its own line to match style elsewhere, such as here: https://github.com/astory/RankPanda/blob/master/RankPanda/RankLocation.py

(Please also do this for the rest of the docstrings)

thanks so much for moving these to docstrings though!

astory commented on RankPanda/CoreWrapper.py:
```diff
@@ -54,16 +52,21 @@ def GetSong(self):
-            moveInfo.append((curmove.GetNumber(), curmove.GetName()))
-            i = i + 1
-        return moveInfo
+        #(Brady) Trying list comprehension
```
Nit: # TODO(Brady): ...
In general I'm following the https://google-styleguide.googlecode.com/svn/trunk/pyguide.html, and have been since I've been editing the code.

You might want to consider using your github name, but Brady is fine.

astory commented on RankPanda/CoreWrapper.py:
```diff
@@ -54,16 +52,21 @@ def GetSong(self):
-        return moveInfo
+        #(Brady) Trying list comprehension
+        return [(curmove.GetNumber(), curmove.GetName()) for curmove in listOfMoves]
+        #Original Code
```
No need for original code, just delete it.  If someone wants history they can use git :)

astory commented on RankPanda/CoreWrapper.py:
```diff
@@ -76,29 +79,37 @@ def GetCurrentMove(self):
     # List of all ranks in current move
     # [(id0, name0, location0), (id1, name1, location1)]
-    def GetRanks(self):
+	#(Brady) Merged next method with this
```
Same deal, don't need to document the history.  Please remove all these notes and all the commented-out code.

astory commented on RankPanda/CoreWrapper.py:
```diff
@@ -76,29 +79,37 @@ def GetCurrentMove(self):
+
+    #(Brady) merged with prev with optional args.  Keeping this so I dont have to change all the method calls.
     # Same as above but now you input the move number.
     def GetRanksGivenMove(self, moveNumber):
```
Is this obsolete now?  Can it be removed?

astory commented on RankPanda/CoreWrapper.py:
```diff
@@ -113,12 +124,15 @@ def GetSelectedRanks(self):
-            i = i + 1
-        return allSelectedRankInfo
+		#(Brady) List comp.
+        return [(r.GetID(), r.GetName(), r.GetEndLocation(), r.GetLabelLocation()) for r in allSelectedRanks]
```
Please wrap to 80 characters.  It's not worth fixing existing code, but if you touch a line, wrap it at 80.

astory commented on RankPanda/CoreWrapper.py:
```diff
@@ -775,6 +789,7 @@ def RankDeleteSpline(self, rankID, number):
             return None
         self._song.currentMove.LookUpID(rankID).DeleteSplinePoint(number)
 
+	#(Brady) TODO: prevent changing name if not on the very first move.
```
Nit: # TODO(Brady):

I'm trying to follow this style guide: https://google-styleguide.googlecode.com/svn/trunk/pyguide.html

astory commented on RankPanda/CubicHermiteSpline.py:
```diff
@@ -19,212 +19,237 @@
 # TODO(astory): convert to module level functions.  None of these actually take
```
Please remove my TODO now that you've done it :)

astory commented on RankPanda/CubicHermiteSpline.py:
```diff
@@ -19,212 +19,237 @@
+# The method we use simply draws a straight line between the point before
+# and the point after, and this becomes the slope of the point.
+# Note that the slopes are dx/dt and dy/dt, not dx/dy.
+def _GetSlope(PointTriple):
```
Is there a reason to start these with an underscore?

astory commented on RankPanda/CubicHermiteSpline.py:
```diff
@@ -19,212 +19,237 @@
+    p1, p2 are a tuple of form (x,y)"""
+    return math.sqrt((p2[0]-p1[0])**2 + (p2[1] - p1[1])**2)
+
+#(Brady) Making it iterative (note it doesnt have underscore...)
```
It's probably better to delete the old code.  If you're nervous about breaking it, write some tests.

astory commented on RankPanda/GUIDialogs.py:
```diff
@@ -137,7 +137,7 @@ def OnAdd(self, event):
         i = 0
         while i < count:
-            val = int(self.inputList.GetItemText(i))
+            val = int(self.inputList.GetItemText(i)) #this gets the first column
```
Nit: please put two spaces between code and #, and one space after the # before the text, also in other comments.

astory commented on RankPanda/GUIDialogs.py:
```diff
@@ -791,7 +804,7 @@ def __init__(self, parent, text, title):
         self.textCtrl = wx.TextCtrl(self, wx.ID_OK, "", style = wx.TE_MULTILINE)
         self.textCtrl.SetValue(text)
-		
+        
```
Could you run `sed -i 's/\s+$// *' in the code directories?  That'll get rid of any trailing whitespace.

brajac709 replied:
So I've never used sed before and I accidentally ran it on all of the files in the code directory, including the fonts and libraries.zip, and it ended up modifying those files as well.  How do I revert changes for just a few specific files?  I have not committed them yet (although I have committed everything else), so how to I get the old versions back in my working directory?  I can't test to see if I broke anything else without them :(

astory commented on RankPanda/GUIDialogs.py:
```diff
@@ -833,25 +846,25 @@ def __init__(self, parent):
         self.directionPanel = wx.BoxSizer(wx.HORIZONTAL)
         self.directionChoices = ['Clockwise', 'Counter-clockwise']
-        self.directionChoicesValues = [0, 1]
+        #self.directionChoicesValues = [0, 1]  #These aren't really needed I dont think
```
Just delete these then.

astory commented on RankPanda/GUIDialogs.py:
```diff
@@ -896,13 +909,14 @@ def OnOK(self, event):
             d.Destroy()
             return   
-        self.output = (self.directionChoicesValues[self.directionRadioBox.GetSelection()], self.pivotChoicesValues[self.pivotRadioBox.GetSelection()] , length)
+        #self.output = (self.directionChoicesValues[self.directionRadioBox.GetSelection()], self.pivotChoicesValues[self.pivotRadioBox.GetSelection()] , length)
```
Please just remove old code.

astory commented on RankPanda/GUIDialogs.py:
```diff
@@ -896,13 +909,14 @@ def OnOK(self, event):
-		
+        self.EndModal(-1)        
+        
+        
```
Please condense multiple empty lines to one.

astory commented on RankPanda/GUIMain.py:
```diff
@@ -276,7 +279,7 @@ def __init__(self, parent):
         self.fieldbar.Bind(wx.EVT_SIZE, self.fieldbar.OnResize)
         self.fieldbar.Bind(wx.EVT_LEFT_DOWN, self.fieldbar.OnLeftClick)
-        self.fieldbar.Bind(wx.EVT_RIGHT_DOWN, self.fieldbar.OnRightClick)
+        self.fieldbar.Bind(wx.EVT_RIGHT_DOWN, self.fieldbar.OnRightClick) #right clicks dont actually do anything, ATM.
```
Please wrap to 80 characters

astory commented on RankPanda/GUIMain.py:
```diff
@@ -357,6 +360,7 @@ def __init__(self, parent):
         self.rankNamePanel = wx.BoxSizer(wx.HORIZONTAL)
         self.rankNamePanel.Add(self.rankNameListPanel, 1, wx.EXPAND)
 
+        #TODO Make this a static Module level or Class level variable.
```
Nit: # TODO(Brady):

astory commented on RankPanda/GUIMain.py:
```diff
@@ -506,7 +511,7 @@ def RefreshMoveList(self):
             item.SetImage(i)
 
-            self.moveSetList.Add(self.field.RenderSet(self.core.GetRanksGivenMove(i)))
+            self.moveSetList.Add(self.field.RenderSet(self.core.GetRanks(moveNumber = i))) #(Brady) changed from GetRanksGivenMove(i) to GetRanks(moveNumber=i)
```
Comment isn't needed.  Good change though.

astory commented on RankPanda/GUIMain.py:
```diff
@@ -811,11 +816,20 @@ def OnCommandAddSpecialButtons(self, event, i):
+        #    if not self.core.IsRankHeld(ranks[n][0]): # if selected rank is not locked, just silently don't do anything
+        #        return
+        #    n=n+1
+#(Brady) Fixed: give a message that some ranks are not locked instead of doing nothing.
```
Don't need old code, comment can just be "Give a message..."

astory commented on RankPanda/GUIMain.py:
```diff
@@ -940,11 +948,19 @@ def OnCommandAdd(self, event):
+        #    if not self.core.IsRankHeld(ranks[n][0]): # if selected rank is not locked, just silently don't do anything
+        #        return
+        #    n=n+1
+		#(Brady) add dialog to tell you its not locked.
```
Comment doesn't need your name attached, get rid of old code.

astory commented on RankPanda/GUIMain.py:
```diff
@@ -1135,7 +1151,7 @@ def OnExport(self, event):
             commandStrings = []
             measureInfo = []
-	    movetexts=[]#[text, overwrite? boolean] tuple list
+            movetexts=[]#[text, overwrite? boolean] tuple list
```
please put space around the #

astory commented on RankPanda/Rank.py:
```diff
@@ -4,6 +4,15 @@
 import pprint
 
+#(Brady) matches command names to function references to call in AddBasicCommand()
+commandHash = {"MarkTime": Commands.MarkTime, "MT": Commands.MarkTime, "ForwardMarch":Commands.ForwardMarch, "FM":Commands.ForwardMarch,
```
Don't need to personalize comment.
Please line wrap to 80 characters.  One-per-line or two-per-line is probably good.

astory commented on RankPanda/Rank.py:
```diff
@@ -255,39 +264,9 @@ def AddBasicCommand(self, commandName, number, length, name):
-            newCommand = Commands.Flatten(length, beginLocation)
+        
+		#(Brady) simplified this by adding a dictionary to hash names to the function
+        newCommand = commandHash[commandName](length, beginLocation) #May want to catch a KeyError in case of typo or something...
```
Please catch the key error and fail with a useful error message like "Unknown command: foo"

astory commented on RankPanda/RankLocation.py:
```diff
@@ -102,10 +102,10 @@ def SetListOfPoints(self, listOfPoints, listOfSlopes):
 
             self._splineFunctions = \
-                CHS.SplineGenerator.GetSplines(self._listOfPoints,
+                CHS.GetSplines(self._listOfPoints,
```
No need for line wrap here and line 226

astory commented on RankPanda/Song.py:
```diff
@@ -369,7 +369,8 @@ def AnimationBegin(self, count):
     def AnimationStep(self):
         if (not self.animating):
-            return (None, 0)
+            print("not self.animating") #(Brady) debug
```
Please remove debug statement.

astory commented on ReadMe.txt:
```diff
@@ -1,18 +1,28 @@
+     http://www.wxpython.org/download.php
+  Install or build them and put them in the site-packages folder of PythonXX\Lib folder  (where XX is the version of python you installed)
+  Python 2.7 may be downloaded from  http://www.python.org/getit/
+  NOTE: RankPanda will not work with Python 3.
```
Somehow you don't have a newline at the end of the file.  Please add it.

astory commented:
This looks great.  Most of my comments are stylistic.  Let's follow https://google-styleguide.googlecode.com/svn/trunk/pyguide.html

brajac709 committed:
Style fixes + more iterative

Made iterative versions of  _GetLengthHelper and
_GetTValueAtLengthHelper.   Baseline tests show that they do the same
thing as recursive versions but more tests may be needed.

brajac709 committed:
Style fixes + more iterative

Made iterative versions of  _GetLengthHelper and
_GetTValueAtLengthHelper.   Baseline tests show that they do the same
thing as recursive versions but more tests may be needed.

brajac709 committed:
Style fixes + more iterative - fix

Made iterative versions of  _GetLengthHelper and
_GetTValueAtLengthHelper.   Baseline tests show that they do the same
thing as recursive versions but more tests may be needed.

brajac709 commented:
I think I covered mostly everything.  Let me know if you find something I missed.  I still want to keep the redundant functions in CubicHermiteSpline.py because my iterative versions have not been fully tested yet.  I had made those changes a while ago without committing them so my git history was kind of messed up when I did some of this.  

I'd say that the most recent commit is not 100% ready yet though: see the sed comment....  But I'll modify it once I get the original other files back.  Just let me know if you find anything really bad that happened...

brajac709 committed:
Merge branch 'brady_pythonify' of https://github.com/brajac709/RankPanda into brady_pythonify

Conflicts:
	RankPanda/RankLocation.py

brajac709 committed:
mend

brajac709 commented:
ok.... so for some reason git didnt want to tell me about conflicts and was doing some weird things, so this commit history looks really messy.  Also something strange happened to all the newlines, so the diffs look pretty bad too....  I think it works though.  So let me know if there's any other style things or ways to reconcile this messiness....
